monkestation/code/modules/admin/battle_royale/battle_royale_controller.dm

```diff
@@ -0,0 +1,574 @@
+
+#define COIN_PRIZE "Coins"
+#define PLAYER_HEALTH_VALUE 150
+//Battle royale controller, IF THERE ARE MULTIPLE OF THESE SOMETHING HAS GONE VERY WRONG, should probably convert into an SS
+/datum/battle_royale_controller
+	///Is this controller active and processing
+	var/active = FALSE
+	///How long has this datum been active for
+	var/active_for = 0
+	///list of our /datum/battle_royale_data datums
+	var/alist/data_datums
+	///list of all our players assigned "antag" datums
+	var/list/players = list()
+	///Tracker var for what data in data_datums we should use next
+	var/next_data_datum_value = 1
+	///Ref to the /datum/battle_royale_data we are currently using
+	var/datum/battle_royale_data/current_data
+	///Ref to our barrier controller
+	var/datum/royale_storm_controller/storm_controller = new
+	///Assoc list of prizes for the winner
+	var/list/prizes = list(COIN_PRIZE = 0,
+						HIGH_THREAT = 0,
+						MEDIUM_THREAT = 0,
+						LOW_THREAT = 0)
+	///What is the expected time for the entire station to be covered in storms
+	var/max_duration = 15 MINUTES
+	///A ref to our signup tracker, cant just use a poll because it breaks at numbers this high
+	var/obj/effect/battle_royale_signup/signup_tracker
+	///The list of turfs valid for us to spawn players and loot on
+	var/list/spawnable_turfs = list()
+	///Assoc list of loot tables to use
+	var/static/list/loot_tables = list(COMMON_LOOT_TABLE = GLOB.royale_common_loot,
+									UTILITY_LOOT_TABLE = GLOB.royale_utility_loot,
+									RARE_LOOT_TABLE = GLOB.royale_rare_loot,
+									SUPER_RARE_LOOT_TABLE = GLOB.royale_super_rare_loot,
+									MISC_LOOT_TABLE = GLOB.royale_misc_loot)
+
+/datum/battle_royale_controller/New()
+	. = ..()
+	if(GLOB.battle_royale_controller)
+		message_admins("Battle royale controller datum created with already existing controller, force ending the current royale of the old controller and qdeling it.")
+		QDEL_NULL(GLOB.battle_royale_controller)
+
+	GLOB.battle_royale_controller = src
+	storm_controller.royale_controller = src
+
+/datum/battle_royale_controller/Destroy(force)
+	message_admins("Battle royale controller datum destroyed, force ending it's current royale.")
+	GLOB.battle_royale_controller = null
+	current_data = null
+	QDEL_LIST_ASSOC_VAL(data_datums)
+	QDEL_NULL(storm_controller)
+	STOP_PROCESSING(SSprocessing, src)
+	return ..()
+
+/datum/battle_royale_controller/process(seconds_per_tick)
+	if(!active)
+		message_admins("Battle royale controller attempting to process while inactive, stopping proccessing.")
+		return PROCESS_KILL
+
+	if(!current_data)
+		message_admins("Battle royale controller attempting to process without set current_data, stopping processing.")
+		deactivate()
+		return
+
+	active_for += seconds_per_tick SECONDS
+	if(check_data())
+		current_data = data_datums[next_data_datum_value]
+		next_data_datum_value++
+
+	spawn_loot_pods(calculate_spawned_loot_pods(seconds_per_tick))
+
+	if(SPT_PROB(current_data.rare_drop_prob, 1))
```

Comment:
```suggestion
	if(SPT_PROB(current_data.rare_drop_prob, seconds_per_tick))
```
why use `SPT_PROB` if you're not gonna pass `seconds_per_tick` or delta time into it?